### PR TITLE
[Merged by Bors] - chore: forward port leanprover-community/mathlib#18946

### PR DESCRIPTION
**Pull request by collares**
Forward port leanprover-community/mathlib#18946.
---

* [`data.real.ennreal`@`29cb56a7b35f72758b05a30490e1f10bd62c35c1`..`c1686dff26eaecf4efd4edd141ebf78de309ae80`](https://leanprover-community.github.io/mathlib-port-status/file/data/real/ennreal?range=29cb56a7b35f72758b05a30490e1f10bd62c35c1..c1686dff26eaecf4efd4edd141ebf78de309ae80)
* [`topology.metric_space.hausdorff_distance`@`57ac39bd365c2f80589a700f9fbb664d3a1a30c2`..`c1686dff26eaecf4efd4edd141ebf78de309ae80`](https://leanprover-community.github.io/mathlib-port-status/file/topology/metric_space/hausdorff_distance?range=57ac39bd365c2f80589a700f9fbb664d3a1a30c2..c1686dff26eaecf4efd4edd141ebf78de309ae80)

* depends on: #3957

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Real/ENNReal.lean [hunk 1]
```diff
@@ -927,8 +927,14 @@ theorem coe_infₛ {s : Set ℝ≥0} : s.Nonempty → (↑(infₛ s) : ℝ≥0
   WithTop.coe_infₛ
 #align ennreal.coe_Inf ENNReal.coe_infₛ
 
-theorem coe_infᵢ {ι} [Nonempty ι] (f : ι → ℝ≥0) : (↑(infᵢ f) : ℝ≥0∞) = ⨅ i, ↑(f i) :=
+theorem coe_supᵢ {ι : Sort _} {f : ι → ℝ≥0} (hf : BddAbove (range f)) :
```
[the remark this answers is not in the document]
Comment (collares) [reply Mathlib/Data/Real/ENNReal.lean:930]: `coe_infᵢ` is missing it indeed, thanks for the nice catch! We didn't add `@[norm_cast]` on `coe_supᵢ ` in mathlib3, though, should I add it here?

[the remark this answers is not in the document]
Comment (collares) [reply Mathlib/Data/Real/ENNReal.lean:932]: Done. I didn't add a note on the basis that this isn't a regression from mathlib3, but I can add it if you have strong opinions on it.

[the remark this answers is not in the document]
Comment (collares) [reply Mathlib/Data/Real/ENNReal.lean:932]: Ah, wait, there are more. Will add them too.

File: Mathlib/Order/ConditionallyCompleteLattice/Basic.lean
Comment (YaelDillies) [inline Mathlib/Order/ConditionallyCompleteLattice/Basic.lean:1]: Could you drop these changes? They are now covered by #3938

Comment (eric-wieser) [reply Mathlib/Order/ConditionallyCompleteLattice/Basic.lean:1]: These don't belong in either this PR or that PR...

Comment (eric-wieser) [reply Mathlib/Order/ConditionallyCompleteLattice/Basic.lean:1]: I just created #3957

Comment (collares) [reply Mathlib/Order/ConditionallyCompleteLattice/Basic.lean:1]: Thank you!

**Review by eric-wieser — APPROVED** [review]
bors merge

Thanks!

**Pull request closed without merging**